core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageAnalyzer.java

```diff
@@ -61,7 +65,7 @@ private void checkStatus(@Nullable String projectKey) {
   public AnalysisResults analyze(ComponentContainer parent, ConnectedAnalysisConfiguration configuration, IssueListener issueListener, ProgressWrapper progress) {
     checkStatus(configuration.projectKey());
 
-    AnalysisContainer analysisContainer = new AnalysisContainer(parent, progress);
+    analysisContainer = new AnalysisContainer(parent, progress);
```

Comment:
I don't think we should do that. The `AnalysisContainer` is intended to be created/disposed for each analysis. You should not keep a reference on it.